EIPS/eip-3290.md

```diff
@@ -0,0 +1,172 @@
+
+6. Vesting
+The most common reason at the moment to lock funds is called "vesting." Imagine that you have just raised a successful ICO and your company still holds a majority of tokens distributed between your team members.
+It would be beneficial to all parties involved to ensure that people's tokens cannot be traded straightaway. If there are no controls in place, anyone might take action by selling all their tokens, cashing out, and quitting the company. This would negatively affect the market price and make all remaining contributors to the project unhappy.
+
+7. DEFI Yielding
+Protect the deposited liquidity or the pool itself.
+
+## Specification
+TBS (Time Based Switch) feature integrates smart contracts and uses events/callbacks to inform the user about the TBS state.
+
+The user creates his TBS by calling the createSwitch function together with four parameters. This returns a signed transaction and puts it in a queue for due execution based on block height or a timestamp.
+
+```
+function createSwitch(uint _time, uint _amount, address_executor, address payable _benefitor)
+    doesntExist
+    checkAmount(_amount)
+    checkTime(_time)
+    public
+    Payable 
+{...}
+```
+
+TBS creation will produce a signed transaction that is put on hold until/if the trigger hits.
+All signed transactions are stored in the TBS queue object in the contract.
+
+Only the executor address is able to trigger a funds transfer when the defined timestamp has expired. The triggered event can be defined as any type of transaction or a specific transaction or blockchain information. Besides that, the creator will determine a percentage of balance dispersed per fallback address.
+
+``` 
+function tryExecuteSwitch(address account)
+    onlyValid(account)
+    onlyExecutorsOrOwner(account)
+    public
+    Payable
+{...}
+```
+
+Required event listener methods for tokens transfer are:
+```approveToken, withdrawToken```
+
+TBS will keep track/store all created switches to be later updated by the owners at will.
+The changeable variables would be:
+
+* Change function fallback addresses as owner
+* Change function condition variables as owner
+* Cancel TBS as an owner
+
+## Rationale
+This EIP standard is a ground layer with default functionality on top of which anyone could build a Dapp or integrate it in his product while allowing for safe and gas-efficient implementations. 
+
+TBS smart contracts should be implemented/imported as a library with only a couple of public functions. Beneficiary logic, notifications, and any other implementation (see motivation) can be abstracted from this standard and implemented on top of it. 
+
+Here are some of the different approaches may include this standard:
+
+* Custom wallet integration for TBS setup
+* (De)centralized notifications/reminders before an action occur.
+* Custom fallback consensus and validation logic
+* Multi TBS setup
+* NFT transfers 
+
```

Comment:
```suggestion
```
Rationale should focus on specific design decisions and their reasoning.  For example, "We chose to return have function X return a value rather than not because Y."